lazyflow/utility/reorderAxesDecorator.py

```diff
@@ -0,0 +1,263 @@
+        old_meth = getattr(cls, meth)
+
+        def guard(fn):
+            @functools.wraps(old_meth)
+            def wrap(self, *args, **kwargs):
+                if self._inner_call:
+                    return fn(self, *args, **kwargs)
+                else:
+                    self._inner_call = True
+                    ret = fn(self, *args, **kwargs)
+                    self._inner_call = False
+                    return ret
+            return wrap
+
+        setattr(cls, meth, guard(old_meth))
+
+    # overwrite class methods with guarded versions thereof
+    for meth in class_methods:
+        old_meth = getattr(cls, meth)
+
+        def guard(fn):
+            @functools.wraps(old_meth)
+            def wrap(self, *args, **kwargs):
+                if self._inner_call:
+                    return fn(*args, **kwargs)
+                else:
+                    self._inner_call = True
+                    ret = fn(*args, **kwargs)
+                    self._inner_call = False
+                    return ret
+            return wrap
+
+        setattr(cls, meth, guard(old_meth))
+
+    # overwrite property methods with guarded versions thereof
+    for meth in property_methods:
+        old_get = getattr(cls, meth).__get__
+
+        def guard(fn):
+            @property
+            def wrap(self):
+                if self._inner_call:
+                    return fn(self)
+                else:
+                    self._inner_call = True
+                    ret = fn(self)
+                    self._inner_call = False
+                    return ret
+            return wrap
+
+        setattr(cls, meth, guard(old_get))
+
+    return cls
+
+
+def reorder(cls):
+    """
+        Decorator function to reorder all input and output channels, to
+        preserve an inner axes order within the operator. Must be used in
+        combination with reorder_options.
+    """
+    if '_reorder_exceptions' not in dir(cls):
+        cls._reorder_exceptions = []
+
+    cls = guard_methods(cls)
+
+    inputSlots = [s.name for s in cls.inputSlots
+                  if s.name not in cls._reorder_exceptions]
+    outputSlots = [s.name for s in cls.outputSlots
+                   if s.name not in cls._reorder_exceptions]
+
+    # change __init__ in order to squeeze in the opReorderAxes ops
+    old_meth = getattr(cls, '__init__')
+
+    def guard(fn):
+        @functools.wraps(old_meth)
+        def wrap(self, *args, **kwargs):
+            self._inner_call = True
+            ret = fn(self, *args, **kwargs)
+
+            self._reorderedOutput = {}
+            for name in outputSlots:
+                self._reorderedOutput[name] = OpReorderAxes(parent=self.parent)
+                self._reorderedOutput[name].AxisOrder.setValue(
+                    cls._output_axes_order)
+                slot = self.__getattribute__(name)
+                self._reorderedOutput[name].Input.connect(slot)
+
+            self._inner_call = False
```

Comment:
Maybe I am not getting the obvious here, but why is `_inner_call = False` from here on? Has this to do with inputs/outputs?